extra/mariabackup/xtrabackup.cc

```diff
@@ -5460,14 +5494,9 @@ static bool xtrabackup_backup_func()
 
 	xb_data_files_close();
 
-	/* Make sure that the latest checkpoint was included */
-	if (metadata_to_lsn > recv_sys.lsn) {
-		msg("Error: failed to copy enough redo log ("
-		    "LSN=" LSN_PF "; checkpoint LSN=" LSN_PF ").",
-		    recv_sys.lsn, metadata_to_lsn);
-		goto fail;
-	}
-
+	ut_ad(!log_copying_running);
+	ut_ad(metadata_to_lsn <= recv_sys.lsn);
+	ut_ad(metadata_last_lsn == recv_sys.lsn);
```

Comment:
This is an important check that ensures that the backup is not corrupt. I would not suggest removing this protection for release mode. Adding debug asserts before this check is fine.

Reply:
This is being checked in `backup_wait_for_commit_lsn()` in `BackupStages::stage_block_commit()`. Had it failed, we’d done `goto fail` and therefore this section of `xtrabackup_backup_func()` would be unreachable.